core/include/gnuradio-4.0/Settings.hpp

```diff
@@ -21,16 +21,39 @@
 namespace gr {
 
 namespace settings {
+
+template<typename T>
+inline constexpr static bool
+isSupportedVectorType() {
+    if constexpr (gr::meta::vector_type<T>) {
+        return std::is_arithmetic_v<typename T::value_type> || std::is_same_v<typename T::value_type, std::string>;
+    } else {
+        return false;
+    }
+}
+
 template<typename T>
 inline constexpr static bool
 isSupportedType() {
-    return std::is_arithmetic_v<T> || std::is_same_v<T, std::string> || gr::meta::vector_type<T> || std::is_same_v<T, property_map>;
+    return std::is_arithmetic_v<T> || std::is_same_v<T, std::string> || isSupportedVectorType<T>() || std::is_same_v<T, property_map>;
+}
+
+template<typename T, typename TMember>
+inline constexpr static bool
+isWritableMember(TMember member) {
+    return traits::port::is_not_any_port_or_collection<T> && !std::is_const_v<T> && is_writable(member) && settings::isSupportedType<T>();
+}
+
+template<typename T, typename TMember>
+inline constexpr static bool
+isReadableMember(TMember member) {
+    return traits::port::is_not_any_port_or_collection<T> && is_readable(member) && settings::isSupportedType<T>();
 }
 } // namespace settings
 
 struct ApplyStagedParametersResult {
-    property_map forwardParameters; // parameters that should be forwarded to dependent child blocks
-    property_map appliedParameters;
+    property_map forwardParameters{}; // parameters that should be forwarded to dependent child blocks
```

Comment:
`{}` should not be necessary as the property_map is default initialised to being empty.